src/App.jsx

```diff
@@ -14,33 +15,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const FIELD_ALPHABETIC = 'alphabetically';
+const FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortBy, setSortBy] = useState(null);
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const getSortedGoods = (orderBy, reverse = false) => {
+    const sorted = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (orderBy === FIELD_ALPHABETIC) {
+      sorted.sort((a, b) => a.localeCompare(b));
+    } else if (orderBy === FIELD_LENGTH) {
+      sorted.sort((a, b) => a.length - b.length);
+    }
+
+    if (reverse) {
+      sorted.reverse();
+    }
+
+    return sorted;
+  };
+
+  const handleSortAlphabetically = () => {
+    const sorted = getSortedGoods(FIELD_ALPHABETIC, isReversed);
+
+    setGoods(sorted);
+    setSortBy(FIELD_ALPHABETIC);
+  };
```

Comment:
```suggestion
  const handleSortAlphabetically = () => {
    setSortBy(FIELD_ALPHABETIC);
  };
```
handler should only change state for sortBy

Reply:
Simply calling setSortB won’t do anything visible, it only changes the sorting state, but no sorting is applied. So with these changes my app is broken.